src/formatDate.js

```diff
@@ -8,7 +8,52 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const partsDate = date.split(fromFormat[3]);
+  let day = '';
+  let month = '';
+  let year = '';
+
+  for (let i = 0; i <= 3; i++) {
```

Comment:
The loop condition `i <= 3` assumes that the `fromFormat` array always has 4 elements. This might not be the case. Consider using `fromFormat.length` to make the loop more flexible.